client/mass/charts.js

```diff
@@ -209,6 +208,34 @@ function getChartTypeList(self, state) {
 			}
 		}
 	]
+	if (state.termdbConfig.allowedTermTypes.includes(TermTypes.METABOLITE_INTENSITY)) {
```

Comment:
here u conditionally add button. this deviates from the existing pattern where all possible buttons are created, and later in line 53 main() are changed in visibility. should keep using this pattern due to its merit of easily show/hide buttons on state change. could be that on surv portal on changing cohort (ccss to sjlife), the MRI button will appear as the MRI datatype only exists in that case.

Reply:
the logic for hiding/showing this button cant be solved checking this.state.supportedChartTypes.includes(d.chartType). There is no metaboliteIntensity chartType. This button decides the chartType dynamically, the same happens with the geneExpression button.

Reply:
it should be fixed when computing supportedChartTypes around line 503 of termdb.server.init.js

Reply:
do you mean I should create a fake chartType?